Added Script to change OpenMRS versions in all Bahmni repositories

jtcwang opened the pull request:
This PR adds a script to change the dependent version of OpenMRS of all `pom.xml` in the following folders (assuming they are at the same directory as `bahmni-environment` folder).

* bahmni-core
* openmrs-distro-bahmni
* openmrs-module-bedmanagement

The search for `pom.xml` in these folders are recursive.

More info is in [this](https://trello.com/c/K790AKd0/1-script-to-change-openmrs-versions-in-all-bahmni-repositories) ticket

gsluthra commented:
Thank you very much for the Pull request! I have requested the team to review and merge. The build will pass next time you send the pull request (I fixed travis ci script).

jtcwang commented:
Cool. Do I need to do anything?

shruthidipali commented:
@jtcwang Can you move this to the scripts folder please?

vinayvenu commented:
sed -i '' (quoted empty string) is required to get rid of the backup files that sed creates (Tested on Yosemite).

rahul080327 commented:
@jtcwang like @shruthidipali said, could you put it in the scripts folder? Aldo the scripts relative paths need to be changed. Lines 21-24 needs change when run from the scripts folder. Use the base folder path as a parameter if you have to instead of using the scripts location as base directory. The extra backup files created by sed too needs clearing.

jtcwang commented:
Updated with some commits

* Now using perl to do replacements as `sed` behaves differently across OSX and linux and there's no good way to fix that.
* moved to `scripts` folder
* No longer use a hardcoded list of directories to check
* The script now recognizes its own location. Search path defaults to `../..` of `scripts` folder. (So it behaves nicely if your `bahmni-environment` folder are siblings of other bahmni repositories). User can supply their own path for recursive search and replace
* Shows what files are checked

rahul080327 commented:
@jtcwang,
After running the script on a Mac OSX 10.11.1 El Capitan,
```xml
<properties>
    <openMRSVersion>1.12.0-SNAPSHOT</openMRSVersion>
</properties>
```
became
```xml
<properties>
        .13</openMRSVersion>
</properties>
```
on all repositories when the script was executed from the bahmni-environment directory as
``` sh scripts/update-openmrs-version.sh 1.13 ```

jtcwang commented:
Can repro. Will debug this.

jtcwang commented:
Updated. I've tested this with both relevant and weird strings and it seems to work fine. Is it working ok for you?

brookingcharlie commented:
If we assume that Python will be installed, something like this could do the replacement:

```
import re
import sys

version = sys.argv[1]

for line in sys.stdin:
  replaced = re.sub(
    '<openMRSVersion>[^<]*</openMRSVersion>',
    '<openMRSVersion>' + version + '</openMRSVersion>',
    line
  )
  sys.stdout.write(replaced)
```

Example usage:

```
$ python re.py '1.11-SNAPSHOT' <<EOF
> <something>
>   <openMRSVersion>1.10</openMRSVersion>
>   <somethingElse />
> </something>
> EOF
<something>
  <openMRSVersion>1.11-SNAPSHOT</openMRSVersion>
  <somethingElse />
</something>
```

aj-jaswanth commented:
@jtcwang Can you squash all 4 commits into one?

jtcwang commented:
Done :smile:

brookingcharlie commented:
Nice work :)
